rosserial_client/src/ros_lib/ros/node_handle.h

```diff
@@ -488,20 +491,33 @@ namespace ros {
     private:
       bool param_recieved;
       rosserial_msgs::RequestParamResponse req_param_resp;
+      rosserial_msgs::PushParamResponse push_param_resp;
 
       bool requestParam(const char * name, int time_out =  1000){
         param_recieved = false;
         rosserial_msgs::RequestParamRequest req;
         req.name  = (char*)name;
         publish(TopicInfo::ID_PARAMETER_REQUEST, &req);
-        uint16_t end_time = hardware_.time() + time_out;
+        uint32_t end_time = hardware_.time() + time_out;
         while(!param_recieved ){
           spinOnce();
           if (hardware_.time() > end_time) return false;
         }
         return true;
       }
 
+      bool pushParam(const rosserial_msgs::PushParamRequest& req, const int time_out = 1000){
+        push_param_resp.success = false;
+
+        publish(TopicInfo::ID_PARAMETER_PUSH, &req);
+        uint32_t end_time = hardware_.time() + time_out;
+        while(!push_param_resp.success){
+          spinOnce();
+          if (hardware_.time() > end_time) return false;
```

Comment:
Not yet - you're right, it would be nice to look into the round trip time on this. Want me to grab that before we merge this, or are you okay saving it for later?